src/utils/events.ts

```diff
@@ -53,43 +54,52 @@ export class Events {
     // first we check for `EventsRecordsFilter` fields for conversion
     // otherwise it is `EventsMessageFilter` fields for conversion
     for (const filter of filters) {
-      if (this.isRecordsFilter(filter)) {
-        eventsQueryFilters.push(Records.convertFilter(filter));
-      } else {
-        eventsQueryFilters.push(this.convertFilter(filter));
+      // extract the protocol tag filter from the incoming event record filter
+      // this filters for permission grants, requests and revocations associated with a targeted protocol
+      // since permissions are their own protocol, we add an additional tag index when writing the permission messages, so we can filter on it here
```

Comment:
Nit:
```suggestion
      // since permissions are their own protocol, we added an additional tag index when writing the permission messages, so we can filter for permission records here
```